src/ledger/LedgerHashUtils.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+// Copyright 2018 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "crypto/ByteSliceHasher.h"
+#include "xdr/Stellar-ledger.h"
+#include <functional>
+
+// implements a default hasher for "LedgerKey"
+template <> class std::hash<stellar::LedgerKey>
+{
+  public:
+    size_t
+    operator()(stellar::LedgerKey const& lk) const
+    {
+        size_t res;
+        switch (lk.type())
+        {
+        case stellar::ACCOUNT:
+            res = stellar::shortHash::computeHash(
+                stellar::ByteSlice(lk.account().accountID.ed25519().data(), 8));
+            break;
+        case stellar::TRUSTLINE:
+            res = stellar::shortHash::computeHash(stellar::ByteSlice(
+                lk.trustLine().accountID.ed25519().data(), 8));
+            res ^= lk.trustLine().asset.type();
```

Comment:
This doesn't seem desirable. Every trust line owned by the same account with the same asset type will hash to the same value. So creating an offer between two `ASSET_TYPE_CREDIT_ALPHANUM4` assets will lead to a guaranteed collision, for example, since both trust lines will have the same hash. This probably needs to include at least some bytes from both the asset code and the issuer.

Reply:
good point asset.type is not what I wanted to use here